web-ui: improvements on block details view (#19)

mario128mex opened the pull request:
### Problem

The transactions for the block view weren't showing useful information and all the transactions for the block were requested

### Solution

Modify the view to add this information and limit the number of transactions requested (add a pagination)

### Result
<img width="1440" alt="captura de pantalla 2018-07-19 a la s 14 18 57" src="https://user-images.githubusercontent.com/6043102/42965317-40bbbac4-8b5f-11e8-8356-399e5ab81c33.png">

<img width="431" alt="captura de pantalla 2018-07-19 a la s 14 19 41" src="https://user-images.githubusercontent.com/6043102/42965322-44457c98-8b5f-11e8-9f99-15c9ebec829f.png">

<img width="428" alt="captura de pantalla 2018-07-19 a la s 14 20 06" src="https://user-images.githubusercontent.com/6043102/42965334-488bbae2-8b5f-11e8-9fe3-ee3d8c6efaab.png">

AlexITC commented:
I see the following issues:
- Displaying the blockhash in the transaction list is useless, all of them should belong to the displayed block
- Where are the pagination controls?
- Displaying the value with a `+` is not that useful, the block is not receiving, nor spending coins.

Thanks.

AlexITC commented on web-ui/src/app/components/block-details/block-details.component.ts:
```diff
@@ -26,7 +36,9 @@ export class BlockDetailsComponent implements OnInit {
 
   ngOnInit() {
-    this.route.params.forEach(params => this.onBlockhash(params['query']));
+    this.blockhash = this.route.snapshot.paramMap.get('query');
```
Why are you changing this?

Load a block on the UI and click the next or previous block links to see why the previous line is required.

AlexITC commented on web-ui/src/app/components/block-details/block-details.component.ts:
```diff
@@ -26,7 +36,11 @@ export class BlockDetailsComponent implements OnInit {
+    this.route.params.forEach((params) => {
+      this.blockhash = params['query'];
+      this.onBlockhash(this.blockhash);
+      this.getPage(this.currentPage);
```
Shouldn't we clear the current values first? let's say that you navigate to the next block from the UI and the server is down, it will keep showing outdated information on the block details and the transactions.

Also, if we move to another block and we are seeing transactions from page 2, does it makes sense to keep viewing transaction from page 2 on the new block?

mario128mex committed:
web-ui: improvements on block details view (#19)

transactions for the block are now retrieved separately, the
transactions table is paginated and shows more info and
finally the view now is mobile friendly

AlexITC commented on web-ui/src/app/components/block-details/block-details.component.html:
```diff
@@ -4,260 +4,288 @@
+                <td>
+                  <a routerLink="/transactions/{{item.id}}">{{item.id | slice:0:35}}...</a>
+                </td>
+                <td>{{item.time * 1000 | date:'MMMM d, y, h:mm:ss a'}}</td>
```
use the pipe please.

AlexITC commented on web-ui/src/app/components/block-details/block-details.component.ts:
```diff
@@ -26,11 +34,24 @@ export class BlockDetailsComponent implements OnInit {
 
   ngOnInit() {
-    this.route.params.forEach(params => this.onBlockhash(params['query']));
+    this.route.params.forEach((params) => {
```
the idea for the `onBlockhash` function is to react to a new blockhash event, please move the logic there.
